src/rgw/rgw_rest_pubsub.cc

```diff
@@ -76,25 +123,41 @@ class RGWPSCreateTopicOp : public RGWOp {
   rgw_pubsub_dest dest;
   std::string topic_arn;
   std::string opaque_data;
-  
+  std::string policy_text;
+
   int get_params() {
     topic_name = s->info.args.get("Name");
     if (topic_name.empty()) {
       ldpp_dout(this, 1) << "CreateTopic Action 'Name' argument is missing" << dendl;
       return -EINVAL;
     }
 
+    // Remove the args that are parsed, so the push_endpoint_args only contains
```

Comment:
why is this needed?

Reply:
> why is this needed?

so currently without removing the args, all the attributes get set into the push_endpoints_args and while listing the topic all the params show twice, once against their struct and once inside the push_endpoint_args.
for example, push_endpoint value is shown in dest.push_endpoint as well as dest.push_endpoint_args
we could exclude them explicitly below or just remove them here.